src/main/java/org/wso2/carbon/connector/KafkaConnectionPool.java

```diff
@@ -90,21 +67,21 @@ private synchronized boolean checkIfConnectionPoolIsFull(MessageContext messageC
      * The ProducerConfig class encapsulates the values required for establishing the connection with brokers such
      * as the broker list, message partition class, serializer class for the message, and partition key,etc.
      *
-     * @param messageContext the message context
-     * @return the connection
+     * @param messageContext the message context.
+     * @return the connection.
      */
-    private Producer<String, String> createNewConnectionForPool(MessageContext messageContext) {
+    private static KafkaProducer createNewConnectionForPool(MessageContext messageContext) {
         KafkaConnection kafkaConnection = new KafkaConnection();
         return kafkaConnection.createNewConnection(messageContext);
     }
 
     /**
      * Get the connection from connection pool.
      *
-     * @return the connection
+     * @return the connection.
      */
-    public synchronized Producer<String, String> getConnectionFromPool() {
-        Producer<String, String> connection = null;
+    public static synchronized KafkaProducer<String, String> getConnectionFromPool() {
+        KafkaProducer<String, String> connection = null;
```

Comment:
yes, I did the research on concurrentLinkedQueue and when I use those I need to implement all the methods for KafkaProducer but when I search on vector It is more thread safe implementation by its own.
https://stackoverflow.com/questions/23246059/java-vector-thread-safety